assets/js/googlesitekit/datastore/forms/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * core/forms Data store: values
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Data from 'googlesitekit-data';
+import invariant from 'invariant';
+import { STORE_NAME } from './constants';
+
+// Actions
+const SET_FORM_VALUES = 'SET_FORM_VALUES';
+
+export { STORE_NAME };
+
+export const INITIAL_STATE = {
+};
+
+export const actions = {
+	setValues( formName, formData ) {
+		invariant( formName, 'form name is required.' );
+
+		return {
+			payload: { formName, formData },
+			type: SET_FORM_VALUES,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_FORM_VALUES: {
+			const { formName, formData } = payload;
+
+			return {
+				...state,
+				[ formName ]: formData,
```

Comment:
This isn't in the AC, but I believe the intention here is to merge the given `formData` with any existing `formData` for that form. That way you don't need to pass the complete state of the entire form every time, but can also use partial objects, similar to [`setState`](https://reactjs.org/docs/react-component.html#setstate).

@tofumatt - can you confirm this is the intended behavior?

One important detail here is that any time we add data into the store from an object, it's important that we always add that to the store _as a new object_. This usually is done using a spread like we do with `state` itself, but the important thing is that the data which is in the store no longer has a reference _outside_ the store where it could be mutated by reference. Spreading an object only makes shallow copies, so it isn't perfect, but it's a reasonable precaution that should always be taken in this context.

Reply:
Ah, yes, that is 100% the intention. You're right—I didn't explicitly set that out in the AC, and it's implied because of the intended use for the store, but I should have mentioned it.

So yes, all existing data should be kept when `setValues()` is dispatched. It should, as @aaemnnosttv mentioned, basically behave like React's `setState()`. 👍